### PR TITLE
[tests-only][full-ci] added test coverage for sharing file between spaces via secure view role

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
Adding test scenario for sharing file via secure roles among the spaces
```feature
Scenario: user copies a file from share space with secure viewer role to personal space
Scenario Outline: user copies a file from share space with secure viewer role to project space with different role
Scenario Outline: user copies a file from share space with different role to share space with role viewer or space viewer

```

## Related Issue

- Part of https://github.com/owncloud/ocis/issues/9334

## How Has This Been Tested?

- test environment:
- locally and CI
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 1]
```diff
@@ -161,20 +161,23 @@ Feature: copy file
...
-      | space           | Personal  |
-      | sharee          | Alice     |
-      | shareType       | user      |
-      | permissionsRole | Viewer    |
+      | resource        | testshare          |
+      | space           | Personal           |
+      | sharee          | Alice              |
+      | shareType       | user               |
+      | permissionsRole | <permissions-role> |
     When user "Alice" copies file "/project.txt" from space "Project" to "/testshare/project.txt" inside space "Shares" using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:169]: IMO, we don't have to test with secure-view here. we want to test the reverse (from Secure-view share to Project space)

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 2]
```diff
@@ -228,19 +231,23 @@ Feature: copy file
...
-      | sharee          | Alice     |
-      | shareType       | user      |
-      | permissionsRole | Viewer    |
+      | resource        | testshare          |
+      | space           | Personal           |
+      | sharee          | Alice              |
+      | shareType       | user               |
+      | permissionsRole | <permissions-role> |
     And user "Alice" has uploaded file with content "personal content" to "/personal.txt"
     When user "Alice" copies file "/personal.txt" from space "Personal" to "/testshare/personal.txt" inside space "Shares" using the WebDAV API
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:243]: IMO, we don't have to test with secure-view here. we want to test the reverse (from Secure-view share to Personal space)

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 3]
```diff
@@ -264,6 +271,20 @@ Feature: copy file
...
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has uploaded file with content "testshare content" to "/testshare/testshare.txt"
+    And user "Brian" has sent the following resource share invitation:
+      | resource        | testshare     |
+      | space           | Personal      |
+      | sharee          | Alice         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
+    When user "Alice" copies file "/testshare/testshare.txt" from space "Shares" to "/testshare.txt" inside space "Personal" using the WebDAV API
+    Then the HTTP status code should be "500"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:285]: Please, report this

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Comment by S-Panta** [comment]
Copying file from Secure view share to personal space returns 500 status code. 
Related issue for that: https://github.com/owncloud/ocis/issues/9482

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature (hunk 3 above)
Comment (SagarGi) [reply tests/acceptance/features/apiSpacesShares/copySpaces.feature:285]: May be this should also go to blocked until it gets fixed? @S-Panta

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 4]
```diff
@@ -150,7 +150,7 @@ Feature: copy file
       | Space Viewer |
 
 
-  Scenario Outline: user copies a file from project space with different role to Shares with viewer role
+  Scenario Outline: user copies a file from project space with different role to Shares with viewer and secure viewer role
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:153]:
```suggestion
  Scenario Outline: user copies a file from project space with different role to Shares with viewer or secure viewer role
```

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 5]
```diff
@@ -228,19 +231,23 @@ Feature: copy file
     And for user "Alice" the content of the file "/testshare/personal.txt" of the space "Shares" should be "personal content"
 
 
-  Scenario: user copies a file from personal space to share space with role viewer
+  Scenario Outline: user copies a file from personal space to share space with role viewer and secure viewer
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:234]:
```suggestion
  Scenario Outline: user copies a file from personal space to share space with role viewer or secure viewer
```

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 6]
```diff
@@ -264,6 +271,20 @@ Feature: copy file
...
+    And user "Brian" has created folder "/testshare"
+    And user "Brian" has uploaded file with content "testshare content" to "/testshare/testshare.txt"
+    And user "Brian" has sent the following resource share invitation:
+      | resource        | testshare     |
+      | space           | Personal      |
+      | sharee          | Alice         |
+      | shareType       | user          |
+      | permissionsRole | Secure viewer |
+    When user "Alice" copies file "/testshare/testshare.txt" from space "Shares" to "/testshare.txt" inside space "Personal" using the WebDAV API
+    Then the HTTP status code should be "403"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:285]: check that the file doesn't appear in the personal space

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 7]
```diff
@@ -293,6 +309,33 @@ Feature: copy file
...
+      | permissionsRole | Secure viewer |
+    When user "Alice" copies file "/testshare/testshare.txt" from space "Shares" to "/testshare.txt" inside space "Project" using the WebDAV API
+    Then the HTTP status code should be "403"
+    And for user "Alice" the space "Project" should not contain these entries:
+      | /testshare.txt |
+    Examples:
+      | space-role   |
+      | Manager      |
+      | Manager      |
+      | Space Editor |
```
Comment (phil-davis) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:336]:
```suggestion
      | Manager      |
      | Space Editor |
```

why is Manager mentioned twice?

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 8]
```diff
@@ -349,32 +392,34 @@ Feature: copy file
       | Viewer           |
 
 
-  Scenario Outline: user copies a file from share space with different role to share space with role viewer
+  Scenario Outline: user copies a file from share space with different role to share space with role viewer or space viewer
```
Comment (phil-davis) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:395]:
```suggestion
  Scenario Outline: user copies a file from share space with different role to share space with role viewer or secure viewer
```

**Review by phil-davis — APPROVED** [review]

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 9]
```diff
@@ -349,32 +391,34 @@ Feature: copy file
...
       | testshare1.txt |
     Examples:
-      | permissions-role |
-      | Editor           |
-      | Viewer           |
+      | permissions-role-1 | permissions-role-2 |
+      | Editor             | Viewer             |
+      | Editor             | Secure viewer      |
+      | Viewer             | Viewer             |
+      | Secure viewer      | Secure viewer      |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:421]:
```suggestion
      | Viewer             | Viewer             |
      | Viewer             | Secure viewer      |
      | Secure viewer      | Viewer             |
      | Secure viewer      | Secure viewer      |
```

File: tests/acceptance/features/apiSpacesShares/copySpaces.feature [hunk 10]
```diff
@@ -264,6 +264,22 @@ Feature: copy file
       | Viewer           |
 
 
+  Scenario: user copies a file from share space with secure viewer role to personal space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/copySpaces.feature:267]:
```suggestion
  @issue-<number>
  Scenario: user copies a file from share space with secure viewer role to personal space
```

**Review by phil-davis — APPROVED** [review]

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**